packages/vsx-registry/src/common/vsx-registry-api.ts

```diff
@@ -107,6 +129,56 @@ export class VSXRegistryAPI {
         return fetchText(url);
     }
 
+    /**
+     * Get the latest compatible extension version.
+     * - an extension satisfies compatibility if its `engines.vscode` version is supported.
+     * @param id the extension id.
+     *
+     * @returns the data for the latest compatible extension version if available, else `undefined`.
+     */
+    async getLatestCompatibleExtensionVersion(id: string): Promise<VSXExtensionRaw | undefined> {
+        const extensions = await this.getAllVersions(id);
+
+        for (let i = 0; i < extensions.length; i++) {
+            const extension: VSXExtensionRaw = extensions[i];
+            if (extension.tags?.includes('Built-in')) {
+                return extension;
+            } else if (extension.version === 'latest') {
```

Comment:
Why this check? `latest` is not a valid version id.

Reply:
Right, I believe it was leftover when I was using `allVersions`, and had to perform the additional requests:

```ts
"allVersions": {
    "latest": "https://open-vsx.org/api/redhat/java/latest",
    "preview": "https://open-vsx.org/api/redhat/java/preview",
    "0.73.0": "https://open-vsx.org/api/redhat/java/0.73.0",
    "0.72.0": "https://open-vsx.org/api/redhat/java/0.72.0",
    "0.71.0": "https://open-vsx.org/api/redhat/java/0.71.0",
    ...
 ```